app/models/starter_file_group.rb

```diff
@@ -28,17 +28,25 @@ def zip_starter_file_files(user)
     zip_path = File.join('tmp', zip_name + '.zip')
     FileUtils.rm_rf zip_path
     Zip::File.open(zip_path, create: true) do |zip_file|
-      self.files_and_dirs.map do |file|
+      write_starter_files_to_zip(zip_file)
+    end
+    zip_path
+  end
+
+  def write_starter_files_to_zip(zip_file, dir = '')
```

Comment:
use `nil` as a default here since you're checking explicitly for whether this variable is set or not. If you wanted to leave it as the empty string you can just use `File.join(dir, file)` because it will be correct whether `dir` is empty or not (`this//path/` is the same as `/this/path/`)